experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1716,31 +1732,19 @@ def _validate_feature_configs(self, data):
             treatment_branch_errors = []
 
             for feature_value_data in treatment_branch_data["feature_values"]:
-                if application == NimbusExperiment.Application.DESKTOP:
-                    if schema_errors := self._validate_feature_value_against_schema(
-                        feature_value_data["feature_config"],
-                        feature_value_data["value"],
-                        localizations,
-                        min_version,
-                        max_version,
-                    ):
-                        treatment_branch_errors.append({"value": schema_errors})
-                        treatment_branches_errors_found = True
-                        continue
-
-                elif NimbusExperiment.Application.is_mobile(application):
-                    if fml_errors := self._validate_with_fml(
-                        loader,
-                        feature_value_data["feature_config"],
-                        feature_value_data["value"],
-                        min_version,
-                        max_version,
-                    ):
-                        treatment_branch_errors.append({"value": fml_errors})
-                        treatment_branches_errors_found = True
-                        continue
-
-                treatment_branch_errors.append({})
+                if feature_errors := self._validate_feature_value(
+                    application,
+                    feature_value_data["feature_config"],
+                    feature_value_data["value"],
+                    min_version,
+                    max_version,
+                    localizations,
+                    channel,
+                ):
+                    treatment_branch_errors.append({"value": feature_errors})
+                    treatment_branches_errors_found = True
+                else:
+                    treatment_branch_errors.append({})
```

Comment:
Thanks @brennie, I really like this change as it also remove our code dependency based on application type, instead of saying` is mobile` or `is desktop` or` is web`